game-core/src/main/java/games/strategy/triplea/TripleA.java

```diff
@@ -149,14 +145,7 @@ private static void connectPlayers(final Set<IGamePlayer> players, final TripleA
   }
 
   @Override
-  public IUnitFactory getUnitFactory() {
-    return new IUnitFactory() {
-      private static final long serialVersionUID = 5684926837825766505L;
-
-      @Override
-      public Unit createUnit(final UnitType type, final PlayerID owner, final GameData data) {
-        return new TripleAUnit(type, owner, data);
-      }
-    };
+  public UnitFactory getUnitFactory() {
+    return TripleAUnit::new;
```

Comment:
At first, this looks weird because you'd expect to see a constructor for the factory.
It makes sense when thinking about it a little bit longer, but lambdas really made classic old object factories kinda obsolete.

Reply:
Yeah, if we're never going to have a stateful `UnitFactory`, then we should probably just inline `UnitFactory` into `IGameLoader` by replacing `IGameLoader#getUnitFactory()` with `createUnit()`.

Unfortunately, `IGameLoader` instances are serialized with `GameData`, so I'm not sure if doing that will break compatibility.  I don't think it will because we're not changing any state.  I can pursue this if you think it makes sense.

Reply:
I don't think doing so will break compatibility.
However I don't really know if the factory is being passed around as an object (which really doesn't look like it) because the diff wouldn't compile otherwise, so if you think this would reduce code, then go for it, depending on the diff size in a separate PR.
Feel free to merge in the latter case.